designs/initial.md

```diff
@@ -0,0 +1,60 @@
+## `@rest` directive
+
+### Arguments
+
+- route: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context
+- params: a map of variables to url params
+- method: the HTTP method to send the request via (i.e GET, PUT, POST)
+- type: The GraphQL type this will return
```

Comment:
- Should there be a discussion about namespace collisions/considerations between GraphQL-native provided types and REST provided types?
- Should there be an entry point for providing custom headers or is the expectation that users will provide those at a different level of the stack? -- Specifically, I'm thinking about "Accept-Type"?

Reply:
> Should there be a discussion about namespace collisions/considerations between GraphQL-native provided types and REST provided types
What cases are you thinking of here?

> Should there be an entry point for providing custom headers
Absolutely! I've been thinking we need to abstract some of the work in the http-link so it can be reused, a lot of it can be shared here to support a variety of fetch specific things (like cancellations for instance)